src/tools/cephfs/JournalTool.cc

```diff
@@ -139,6 +139,12 @@ int JournalTool::main(std::vector<const char*> &argv)
   }
  
   auto& fs = fsmap->get_filesystem(role_selector.get_ns());
+  stringstream (rank_str.substr(rank_str.find(':') + 1)) >> rank;
+  if (fs.get_mds_map().is_active(rank)) {
+    derr << "Cannot run cephfs-journal-tool on an active file system!" << dendl;
+    return -CEPHFS_EPERM;
+  }
```

Comment:
so we have completely disabled running journal tool on an active fs, it is fine since it will not cause additional damage but what if the admin wants to really do it since he/she knows what he/she is upto? is it always deadly to run this tool on active fs or can it be run but with extreme caution?

Reply:
> so we have completely disabled running journal tool on an active fs, it is fine since it will not cause additional damage but what if the admin wants to really do it since he/she knows what he/she is upto?

I'l be really afraid of such cluster admins :)

If they *really really* want to, then they can compile their own version of the patched up tool and run at will. I don't see a case where we should allow the tool to run on an active fs. If someone can show me a case, I'll be happy to change my mind.

Reply:
i don't know of any such case, maybe it is fine to go this way and change back in future if a case arises